### PR TITLE
Replace rosidl_generator_x for rosidl_runtime_x

### PR DESCRIPTION
**Pull request by ahcorde**
This PR is related to the changes introduced in this PR ros2/rosidl#442. The full process can be followed here ros2/rosidl#443

`rmw_connext` is using `rosidl_generator_c` headers which are located here: [rosidl_runtime_c](https://github.com/ros2/rosidl/tree/ahcorde/feature/splitted_rosidl_generator/rosidl_runtime_c/include/rosidl_generator_c)

Example os file using these headers 

https://github.com/ros2/rmw_connext/blob/582e69e1609449c09e675fba4103190a8c912d0a/rmw_connext_dynamic_cpp/src/templates.hpp#L22

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

**Comment by ahcorde** [comment]
I made the change of  `rosidl_runtime_c` to `<depend>` because it has a .so/.dylib/dll

File: rmw_connext_cpp/package.xml [hunk 1]
```diff
@@ -17,15 +17,13 @@
...
-  <build_depend>rosidl_generator_cpp</build_depend>
+  <build_depend>rosidl_runtime_cpp</build_depend>
   <build_depend>rosidl_generator_dds_idl</build_depend>
   <build_depend>rosidl_typesupport_connext_c</build_depend>
   <build_depend>rosidl_typesupport_connext_cpp</build_depend>
   <build_depend>rti-connext-dds-5.3.1</build_depend>
 
   <build_export_depend>connext_cmake_module</build_export_depend>
-  <build_export_depend>rosidl_generator_c</build_export_depend>
   <build_export_depend>rosidl_generator_cpp</build_export_depend>
```
Comment (dirk-thomas) [inline rmw_connext_cpp/package.xml]: Why is this still depending on the generator package?

Comment (ahcorde) [reply rmw_connext_cpp/package.xml]: https://github.com/ros2/rmw_connext/pull/399/commits/84893bc4368133086b93054e7048c896e74bb614

**Comment by ahcorde** [comment]
For some reason this failure has appeared in the buildfarm https://ci.ros2.org/job/ci_linux/9920/console:

```
16:55:26 Traceback (most recent call last):
16:55:26   File "/home/jenkins-agent/workspace/ci_linux/ws/src/ros2/rmw_connext/rmw_connext_cpp/bin/apply-patch.py", line 80, in <module>
16:55:26     content_out = apply_patch(content_in, content_patch)
16:55:26   File "/home/jenkins-agent/workspace/ci_linux/ws/src/ros2/rmw_connext/rmw_connext_cpp/bin/apply-patch.py", line 60, in apply_patch
16:55:26     assert s[sl] == line[1:], \
16:55:26 AssertionError: s[323] '
16:55:26 ' != line[1:] '    NDDSUSERDllExport extern struct PRESTypePlugin*
16:55:26 '
 ```

this is the right patch https://github.com/ros2/rmw_connext/pull/399/commits/28b1b0478c379d0beac2500b148ebdedbcb5563a:
```
--- a/rmw_connext_cpp/resources/patch_generated/connext_static_serialized_dataPlugin.h
+++ b/rmw_connext_cpp/resources/patch_generated/connext_static_serialized_dataPlugin.h
@@ -322,6 +322,9 @@
 NDDSUSERDllExport extern struct PRESTypePlugin*
 ConnextStaticSerializedDataPlugin_new(void);

+NDDSUSERDllExport extern struct PRESTypePlugin*
+ConnextStaticSerializedDataPlugin_new_external(struct DDS_TypeCode * external_type_code);
+
 NDDSUSERDllExport extern void
 ConnextStaticSerializedDataPlugin_delete(struct PRESTypePlugin *);
```

Any idea why is this happening? @dirk-thomas @Karsten1987 ?

**Comment by ahcorde** [comment]
By the way I know this an unrelated change, I can split this new change in a new PR

**Comment by dirk-thomas** [comment]
> By the way I know this an unrelated change, I can split this new change in a new PR

Then please do separate it. This has nothing to do with the topic of this PR.

**Comment by ahcorde** [comment]
- Updated all repos involved in this change. 
 - Skipping `ros1_bridge`, `qt_*`, `rqt_*`, and `rviz_*` packages. 
 - Testing against `test_rclcpp` and `rosbag2_tests`.

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10031/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5626/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8161/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9938/)

 - One warning made the builds unstable but there is a ([PR pending](https://gitlab.com/micro-ROS/ros_tracing/ros2_tracing/-/merge_requests/163))

```
CMake Deprecation Warning at /home/jenkins-agent/workspace/ci_linux-aarch64/ws/install/ament_cmake_export_interfaces/share/ament_cmake_export_interfaces/cmake/ament_export_interfaces.cmake:37 (message):
ament_export_interfaces() is deprecated, use ament_export_targets() instead
Call Stack (most recent call first):
CMakeLists.txt:85 (ament_export_interfaces)
``` 

In MacOS
 - projectroot.cppcheck

Windows
 - some failures in test_rclcpp

**Pull request merged**